gcloud/monitoring/group.py

```diff
@@ -0,0 +1,502 @@
+              descendants_of_group=None):
+        """Lists all groups in the project.
+
+        :type client: :class:`gcloud.monitoring.client.Client`
+        :param client: The client to use.
+
+        :type children_of_group: string or None
+        :param children_of_group:
+            Returns groups whose parent_name field contains the group name. If
+            no groups have this parent, the results are empty.
+
+        :type ancestors_of_group: string or None
+        :param ancestors_of_group:
+            Returns groups that are ancestors of the specified group. If the
+            specified group has no immediate parent, the results are empty.
+
+        :type descendants_of_group: string or None
+        :param descendants_of_group:
+            Returns the descendants of the specified group. This is a superset
+            of the results returned by the children_of_group filter, and
+            includes children-of-children, and so forth.
+
+        :rtype: list of :class:`~gcloud.monitoring.group.Group`
+        :returns: A list of group instances.
+        """
+        path = cls.path_helper(client.project)
+        groups = []
+        page_token = None
+
+        while True:
+            params = {}
+
+            if children_of_group is not None:
+                params['childrenOfGroup'] = children_of_group
+
+            if ancestors_of_group is not None:
+                params['ancestorsOfGroup'] = ancestors_of_group
+
+            if descendants_of_group is not None:
+                params['descendantsOfGroup'] = descendants_of_group
+
+            if page_token is not None:
+                params['pageToken'] = page_token
+
+            response = client.connection.api_request(
+                method='GET', path=path, query_params=params)
+            for info in response.get('group', []):
+                groups.append(cls._from_dict(client, info))
+
+            page_token = response.get('nextPageToken')
+            if not page_token:
+                break
+
+        return groups
+
+    @classmethod
+    def _from_dict(cls, client, info):
+        """Constructs a Group instance from the parsed JSON representation.
+
+        :type client: :class:`gcloud.monitoring.client.Client`
+        :param client: A client to be included in the returned object.
+
+        :type info: dict
+        :param info:
+            A ``dict`` parsed from the JSON wire-format representation.
+
+        :rtype: :class:`Group`
+        :returns: A group.
+        """
+        group = cls(client)
+        group._init_from_dict(info)
+        return group
+
+    def _init_from_dict(self, info):
+        """Initialize attributes from the parsed JSON representation.
+
+        :type info: dict
+        :param info:
+            A ``dict`` parsed from the JSON wire-format representation.
+        """
+        self._name = info['name']
+        self._id = group_id_from_name(info['name'])
```

Comment:
You could use `self._name` instead of `info['name']`, though both lookups are constant time?